### PR TITLE
Scalafix for replacing Sink with Pipe

### PR DESCRIPTION
**Pull request by tmciver**
One possible issue with this PR is that there are 2 input and output files associated with this scalafix rule.  Would it be preferred that they be combined into one?

### CODE REVIEW

File: scalafix/input/src/main/scala/fix/Fs2sinkremoval.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/*
+rule = v1
+*/
+package fix
+
+import fs2._
+import fs2.{Pipe, Sink, Stream}
+import fs2.{Sink => X, Stream}
```
Comment (mpilquist) [inline scalafix/input/src/main/scala/fix/Fs2sinkremoval.scala]: Did you intend to add a `X[F, A]` example below?

File: scalafix/input/src/main/scala/fix/Fs2sinkremoval.scala [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+object Fs2sinkremoval {
+
+  private def foo[F[_], A](s: Stream[F, A], sink: Sink[F, A]): Stream[F, Unit] =
+    s.to(sink)
+
+  def bar[F[_], A, B](): Pipe[F, A, B] = ???
```
Comment (mpilquist) [inline scalafix/input/src/main/scala/fix/Fs2sinkremoval.scala]: Should this return `Sink[F, A]`?

Comment (tmciver) [reply scalafix/input/src/main/scala/fix/Fs2sinkremoval.scala]: Yup. Fixed.

File: scalafix/input/src/main/scala/fix/Fs2sinkremoval.scala (hunk 1 above)
Comment (tmciver) [reply scalafix/input/src/main/scala/fix/Fs2sinkremoval.scala]: Not sure about this one.  @monadplus what was the intent of the `Sink` rename?

File: scalafix/rules/src/main/scala/fix/v1.scala [hunk 3]
```diff
@@ -530,13 +530,20 @@ object SinkToPipeRules {
   ).asPatch
 
   def apply(t: Tree)(implicit doc: SemanticDocument): List[Patch] = {
+    //println("Tree.structureLabeled: " + doc.tree.structureLabeled)
+    //val sinkMatcher = SymbolMatcher.normalized("fs2.Sink")
+    val sinkMatcher = SymbolMatcher.exact("fs2/Sink.")
```
Comment (olafurpg) [inline scalafix/rules/src/main/scala/fix/v1.scala]: What happens with `fs2/Sink#`?

Comment (tmciver) [reply scalafix/rules/src/main/scala/fix/v1.scala]: It doesn't match but I'm not sure why.

Comment (olafurpg) [reply scalafix/rules/src/main/scala/fix/v1.scala]: What symbol is printed by `Tree.symbol` of the trees you want to match?

Comment (tmciver) [reply scalafix/rules/src/main/scala/fix/v1.scala]: BTW, what's the difference between `fs2/Sink.` and `fs2/Sink#`?  I don't think the docs are clear on that.

Comment (olafurpg) [reply scalafix/rules/src/main/scala/fix/v1.scala]: the symbol format is documented in the SemanticDB specification https://scalameta.org/docs/semanticdb/specification.html#symbol

`Sink.` is the object symbol and `Sink#` is the class or trait symbol

**Comment by mpilquist** [comment]
Thanks for all the work on this one!

**Pull request merged**